sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -406,6 +406,16 @@ ze_result_t ZeCall::doCall(ze_result_t ZeResult, const char *CallStr,
     return mapError(Result);
 #define ZE_CALL_NOCHECK(Call) ZeCall().doCall(Call, #Call, false)
 
+_pi_device::~_pi_device() {
+  // Destroy all the command lists associated with this device.
+  std::lock_guard<std::mutex> Lock(ZeCommandListCacheMutex);
+  for (ze_command_list_handle_t &ZeCommandList : ZeCommandListCache) {
+    if (ZeCommandList)
+      zeCommandListDestroy(ZeCommandList);
+  }
+  ZeCommandListCache.clear();
```

Comment:
This call of clear() method seems redundant as the whole list is going to be cleared & destroyed after ~pi_device() call.

Reply:
This was not ready for review. I was in the middle of debugging memory leaks.
Anyway, I fixed the issues you pointed out.
Thanks.